### ISSUE TITLE
findByIdAndUpdate / findOneAndUpdate gives null result with Mongoose 4.x

### ISSUE THREAD

**Issue by Rob--W** [mccormicka/Mockgoose#121]
findByIdAndUpdate / findOneAndUpdate gives a `null` result when Mockgoose is used in conjunction with Mongoose 4.x.

Steps to reproduce:

1. Install mockgoose & mongoose:

    ```
npm install mockgoose@2.0.2
npm install mongoose@4.0.4
```

2. Save the following script and run it.

    ```js
#!/usr/bin/env node
'use strict';

// Whether to use old or new Mongoose
var old = false;

var mongoose = require(old ? 'mockgoose/node_modules/mongoose' : 'mongoose');
require('mockgoose')(mongoose);

var SomeSchema = new mongoose.Schema({
    foo: String,
});

SomeSchema.statics.findAndExec = function(id) {
    this.findByIdAndUpdate(id, {
        foo: 'red'
    }, {
        new: true
    }, function(e, result) {
        if (e) throw e;
        console.assert(result !== null, 'result shall not be null');
        console.log('Done!', result);
    }); 
};

var Some = mongoose.model('Some', SomeSchema);

mongoose.connect('mongodb://localhost/whatever', onConnected);

function onConnected() {
    new Some({
        foo: 'bawz',
    }).save(function(er, mod) {
        if (er) throw er; 
        process.nextTick(function() {
            Some.findAndExec(mod._id);
        }); 
    }); 
}
```

Expected result: "Done!, { foo: 'red', _id: 556bfdf1cec13a6d5ce83446, __v: 0 }"  
Actual result: `result` is null, which is apparent from the assertion failure.

```
$ node index.js 

##############################################################
#
#   !!! MONGOOSE WARNING !!!
#
#   This is an UNSTABLE release of Mongoose.
#   Unstable releases are available for preview/testing only.
#   DO NOT run this in production.
#
##############################################################

assert.js:86
  throw new assert.AssertionError({
        ^
AssertionError: result shall not be null
    at Console.assert (console.js:106:23)
    at Query.<anonymous> (/tmp/goo/index.js:21:17)
    at /tmp/goo/node_modules/mongoose/node_modules/kareem/index.js:177:19
    at /tmp/goo/node_modules/mongoose/node_modules/kareem/index.js:109:16
    at process._tickCallback (node.js:355:11)
```

**Comment by tamlyn** [comment]
Does downgrading to 3.8 fix it? See #125.

**Comment by Rob--W** [comment]
@tamlyn Downgrading works (set `old = true;` in the snippet above). That's hardly a solution though.

**Comment by tamlyn** [comment]
No, indeed. However the problem, as I understand it isn't with 4.0 per se. The problem is that it's using different versions. This can be fixed quite simply as detailed in #125 although there may be other issues I haven't thought of.

**Comment by schmittyjd** [comment]
@tamlyn @Rob--W Mongoose 4.X introduced breaking changes to findByIdAndUpdate and similar methods. See here https://github.com/Automattic/mongoose/wiki/4.0-Release-Notes #2262

**Comment by schmittyjd** [comment]
The flag new is now turned off by default

**Comment by winfinit** [comment]
this should be fixed with 5.0
